web/text_layer_builder.js

```diff
@@ -243,9 +246,17 @@ class TextLayerBuilder {
       let isSelected = (isSelectedPage && i === selectedMatchIdx);
       let highlightSuffix = (isSelected ? ' selected' : '');
 
+      // Scroll the selected match into view.
       if (this.findController) {
-        this.findController.updateMatchPosition(pageIdx, i, textDivs,
-                                                begin.divIdx);
+        if (this.findController.selected.matchIdx === i &&
+            this.findController.selected.pageIdx === pageIdx) {
```

Comment:
This might be (yet another) pre-existing issue, but `selected` feels like it should be internal FindController state not directly accessed from the outside in this way.

Can `selected` be made "private" in `PDFFindController`, and instead accessed e.g. through a `get selected() { return this._selected; };` method instead?

Reply:
Done in the new commit series. The only ones that are still public are members that are being assigned to from outside. I do think that this pattern should be changed later on, but I'd rather not do that in this PR to keep its size limited.

Reply:
> I do think that this pattern should be changed later on, but I'd rather not do that in this PR to keep its size limited.

Good call, let's avoid too much scope creep here.